.github/workflows/3rd-party-tests.yml

```
@@ -0,0 +1,88 @@
+name: Test on 3rd party libraries
+
+on:
+  push:
+    branches:
+      - "master"
+      - "sqlalchemy_pipeline"
+    paths-ignore:
+      - "docs/*"
+      - "tools/*"
+  workflow_dispatch:
+
+concurrency:
+  # Cancel older requests of the same workflow in the same branch.
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  sqlalchemy:
+    # linux should be enough to test if everything works.
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.11"
+        sqlalchemy:
+          - git+https://github.com/sqlalchemy/sqlalchemy.git#egg=SQLAlchemy
+          - sqlalchemy
+        impl:
+          - c
+          - python
+
+    env:
+      PSYCOPG_IMPL: ${{ matrix.impl }}
+      DEPS: ./psycopg[test] ./psycopg_pool pytest pytest-xdist
+
+    services:
+      pg:
+        image: postgres:15
+        env:
+          POSTGRES_PASSWORD: password
+          POSTGRES_DB: test
+        ports:
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Setup PG
+        env:
+          PGPASSWORD: password
+        run: |
+          run_sql="psql -h 127.0.0.1 -d test -U postgres -c "
+          $run_sql 'SELECT version();'
+          $run_sql 'CREATE SCHEMA test_schema; CREATE SCHEMA test_schema_2;CREATE EXTENSION hstore;'
+
+      - name: Include psycopg-c to the packages to install
+        if: ${{ matrix.impl == 'c' }}
+        run: |
+          echo "DEPS=$DEPS ./psycopg_c" >> $GITHUB_ENV
+
+      - name: Install pycopg packages
+        run: pip install $DEPS
+
+      - name: Setup and install sqlalchemy
+        # TODO: remove pre once v2 is stable
+        run: |
+          pip download --pre --no-deps --no-binary :all: ${{ matrix.sqlalchemy }}
+          mkdir sa_home
+          if [[ ${{ matrix.sqlalchemy }} = "sqlalchemy" ]]
+          then
+            extract="tar -C sa_home -xzf"
+          else
+            extract="unzip -d sa_home -q"
+          fi
+          $extract $( ls SQLAlchemy* )
+          cd sa_home/$( ls sa_home )
+          pip install .
+
+      - name: Run sqlalchemy tests
+        run: bash -c "cd sa_home/$( ls sa_home ) &&
+          pytest -n 2 -q --dburi postgresql+psycopg://postgres:password@127.0.0.1/test --backend-only --dropfirst
+          "
```

Multiline would be more readable:

```suggestion
        run: |
          cd sa_home/$( ls sa_home )
          pytest -n 2 -q --backend-only --dropfirst \
             --dburi postgresql+psycopg://postgres:password@127.0.0.1/test

```

---

I was experimenting, since GitHub doesn't show the output of the test until they are done for some reason.

I wanted to try another alternative to see if it works, if it doesn't I'll use what you suggested